src/app/app.component.spec.ts

```diff
@@ -1,17 +1,52 @@
-import { TestBed } from '@angular/core/testing';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 import { AppComponent } from './app.component';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {WeatherServiceService} from './weather-service.service';
+import {of} from 'rxjs';
+import {WeatherDataComponent} from './weather-data/weather-data.component';
 
 describe('AppComponent', () => {
+
+  let component: AppComponent;
+  let fixture: ComponentFixture<AppComponent>;
+  let service: WeatherServiceService;
+  let httpTestingController: HttpTestingController;
+  let mockData = 1;
+  let mockResult:any = [];
+
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
+        HttpClientTestingModule,
         RouterTestingModule
       ],
       declarations: [
-        AppComponent
+        AppComponent,
+        WeatherDataComponent
       ],
+      providers:[
+        WeatherServiceService
+      ]
     }).compileComponents();
   });
 
+  beforeEach(() => {
+    service = TestBed.inject(WeatherServiceService);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    fixture = TestBed.createComponent(AppComponent);
+    component = fixture.componentInstance;
+  })
+
+  it('getData', async () => {
```

Comment:
i'd create a describe block called 'methods' and then another inside it called 'getData'
then the tests within that are always in the format: it('Should do thing thing i am expecting')

that way in the test reports it'll come out as:
app.component > methods > getData > Should do a thing
rather than
app.component > getData
as it is currently